catkin_ws/src/vision/src/object_detection_utils.py

```diff
@@ -246,8 +263,8 @@ def getObjectPosition(pixel_x, pixel_y, img_height, img_width, dist_from_camera=
 
 def measureBuoyAngle(depth_cropped):
     depth_cropped = clean_depth_error(depth_cropped, error_threshold=0.2)
-    rows, _ = depth_cropped.shape
-    left_half, right_half = depth_cropped[:rows/2, :], depth_cropped[rows/2:, :]
+    _, cols = depth_cropped.shape
+    left_half, right_half = depth_cropped[:, :int(cols/2)], depth_cropped[:, int(cols/2):]
```

Comment:
This function (not related to this pull request but I just noticed) should give an exact angle, not just a 1 or -1. You can take the same math as elie did for the gate rotation (it's still in his branch) and I think it would pretty much the same, but would need to filter out the background first

Reply:
what?